google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/RateLimitingServerStreamingCallable.java

```diff
@@ -88,44 +89,164 @@ public void call(
       ((BigtableTracer) context.getTracer())
           .batchRequestThrottled(stopwatch.elapsed(TimeUnit.NANOSECONDS));
     }
-    RateLimitingResponseObserver innerObserver =
-        new RateLimitingResponseObserver(limiter, lastQpsChangeTime, responseObserver);
+    RateLimitingResponseObserver innerObserver = new RateLimitingResponseObserver(responseObserver);
     innerCallable.call(request, innerObserver, context);
   }
 
+  /** A rate limiter wrapper class that can be disabled. */
+  static class ConditionalRateLimiter {
+
+    private final AtomicBoolean enabled = new AtomicBoolean(false);
+
+    private final RateLimiter limiter;
+
+    // This is the next time allowed to change QPS or disable rate limiting.
+    private final AtomicReference<Instant> nextRateUpdateTime =
+        new AtomicReference<>(Instant.now());
+
+    public ConditionalRateLimiter(long defaultQps) {
+      limiter = RateLimiter.create(defaultQps);
+    }
+
+    /**
+     * Works the same way with {@link RateLimiter#acquire()} except that when the rate limiter is
+     * disabled, {@link ConditionalRateLimiter#acquire()} always returns 0.0 immediately.
+     */
+    public double acquire() {
```

Comment:
What does the return value mean? and who actually uses it?
Looking at the guava docs it seems like RateLimiter#acquire returns the amount of time spent waiting for throttling. This seems duplicated in the Callable above via the stopwatch.

I think you should pick one of 2 paths:
1. Use the functionality provided to you by guava and document what is being returned. But maybe have your wrapper do something nicer like return a Duration instead of a unit-less double
2. Ignore the guava functionality and drop the return type. But consider moving the stopwatch into the ConditionalRateLimiter

Reply:
Discussed offline. Removed the return type for now.